### ISSUE TITLE
Improve sound handling

### ISSUE THREAD

**Issue by AMDmi3** [AMDmi3/qnetwalk#3]
Having to depend on SDL is strange. Qt multimedia or some simpler library, like maybe sndio, could be used. Also the sound plays with noticeable delay, which should be fixed too.

**Comment by yselkowitz** [comment]
Unfortunately I don't hear any sound, even when enabled, in 1.9.0 when built with Qt6.  1.9.0 with Qt5 is fine though, as is 1.8.0 regardless of Qt version.

**Comment by AMDmi3** [comment]
Are there any messages on console? Is there sound in any other qt6-multimedia consumers?

**Comment by yselkowitz** [comment]
now that you mention it, I see five of these with the affected build:
```
QNetworkRequest::setHeader: QVariant of type QDateTime cannot be used with header Last-Modified
```
I'd have to find another consumer of Qt6Multimedia to test there.

**Comment by AMDmi3** [comment]
The fact that there are five of these does suggest that these are related to attempts to load 5 sounds the game contains. That it mentions http headers is completely unexpected though, I'd never expect `QUrl::fromLocalFile` to involve any networking. Perhaps something related to it is broken or misused.

**Comment by yselkowitz** [comment]
Does any of https://doc.qt.io/qt-6/qtmultimedia-changes-qt6.html apply to the code here?

**Comment by AMDmi3** [comment]
Not really. Should probably read qt code a bit to understand what may be happening here, in the meantime could you try dumping `source` of an effect after its creation in initSoundEffect? Also, which qt version do you have?

**Comment by yselkowitz** [comment]
Sorry, I'm not sure how to go about doing that.  But I can tell you I'm using Qt 6.4.2.

**Comment by AMDmi3** [comment]
Ok, I'm afk till next week, then I'll meke a branch for debugging this.

**Comment by AMDmi3** [comment]
Try `sound-debug` branch please.

**Comment by yselkowitz** [comment]
I'm not seeing any of the debug output with qt6, so it seems it's not getting that far?

**Comment by AMDmi3** [comment]
Are you sure you've built from the right branch and haven't disabled sound support (ENABLE_SOUND)?
